### PR TITLE
Fix: set the value properly for checkbox template

### PR DESCRIPTION
**Pull request by JasonTheAdams**
## Description

#186 added the checkbox template, but it didn't correctly add the necessary value input. I've since added this in https://github.com/impress-org/givewp/pull/6813.

As a note, the `checked` attribute _should not_ be applied since React Hook Form will set that based on whether the default value is equal to the input value.

## Affects

Only the checkbox template

## Pre-review Checklist

-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@jonwaldstein In light of the changes in https://github.com/impress-org/givewp/pull/6813 and the checkbox PR already being merged, this is necessary to properly set the value.

**Comment by JasonTheAdams** [comment]
@pauloiankoski I believe you used the checkbox template in Fee Recovery? Does this change mean you'll need to change something over there?

**Comment by pauloiankoski** [comment]
@JasonTheAdams I do use that component, but it's not part of the React Hook Form system. It's only used to enable the logic, not to pass any value to the backend. For that reason, I don't think we need to change anything on Fee Recovery.

CleanShot 2023-06-02 at 20 23 44

**Comment by JasonTheAdams** [comment]
Interesting, @pauloiankoski. I would expect what you're doing there to throw TypeScript errors. You're not passing the `value` and the `inputProps` aren't complete compared to what it's expecting. I wouldn't change how you're using it, but I'd expect you to have to suppress some TS errors.

**Comment by pauloiankoski** [comment]
@JasonTheAdams I'm probably not receiving any error because there are not types for the Checkbox component in the addon context.

**Comment by JasonTheAdams** [comment]
Ah, I see. Got it. Well, it should still work.

**Comment by JasonTheAdams** [comment]
@jonwaldstein I've resolved the merge conflict and merged the corresponding PR in core. This is ready for review!

File: src/DonationForms/resources/propTypes.ts [hunk 1]
```diff
@@ -35,6 +35,10 @@ export interface RadioFieldProps extends FieldProps {
     options: Array<RadioOption>;
 }
 
+export interface CheckboxProps extends FieldProps {
+    value: string | number;
```
Comment (jonwaldstein) [inline src/DonationForms/resources/propTypes.ts:39]: I think we should add boolean as a type, or will that be the default if value is not provided?

Comment (jonwaldstein) [reply src/DonationForms/resources/propTypes.ts:39]: oh interesting, well I guess that makes sense.

**Review by jonwaldstein — APPROVED** [review]

**Pull request merged**